Failing with:
```
...
          tracing_subscriber.ec056020e7b8cd5e-cgu.07:(.text._ZN4core3ptr42drop_in_place$LT$std..io..error..Error$GT$17hf9da2e6f131ca017E.llvm.1858921144840505084+0x6a): undefined reference to `sdallocx'
          /nix/store/81xsp348yfgmaan9r5055mcdjfw7a8wc-binutils-2.42/bin/ld: tracing_subscriber.ec056020e7b8cd5e-cgu.07:(.text._ZN4core3ptr42drop_in_place$LT$std..io..error..Error$GT$17hf9da2e6f131ca017E.llvm.1858921144840505084+0x85): undefined reference to `sdallocx'
          /nix/store/81xsp348yfgmaan9r5055mcdjfw7a8wc-binutils-2.42/bin/ld: tracing_subscriber.ec056020e7b8cd5e-cgu.07:(.text._ZN4core3ptr42drop_in_place$LT$std..io..error..Error$GT$17hf9da2e6f131ca017E.llvm.1858921144840505084+0xb4): undefined reference to `sdallocx'
          /nix/store/81xsp348yfgmaan9r5055mcdjfw7a8wc-binutils-2.42/bin/ld: /build/source/target/x86_64-unknown-linux-gnu/release/deps/ruff-fe24bed45e65f118.tracing_subscriber-c0d529402cf3a3aa.tracing_subscriber.ec056020e7b8cd5e-cgu.07.rcgu.o.rcgu.o:tracing_subscriber.ec056020e7b8cd5e-cgu.07:(.text._ZN4core3ptr42drop_in_place$LT$std..io..error..Error$GT$17hf9da2e6f131ca017E.llvm.1858921144840505084+0xc4): more undefined references to `sdallocx' follow
...
```

Full log: https://paste.glepage.com/upload/otter-snail-shark
This issue might be related: https://github.com/astral-sh/ruff/issues/9933

--- a/pkgs/by-name/ru/ruff/package.nix
+++ b/pkgs/by-name/ru/ruff/package.nix
@@ -14,13 +14,13 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "ruff";
-  version = "0.6.4";
+  version = "0.6.5";
 
   src = fetchFromGitHub {
     owner = "astral-sh";
     repo = "ruff";
     rev = "refs/tags/${version}";
-    hash = "sha256-AldYWbLtkVtM1sWBCgNym9RZ0QszIh59vQhoysl5/3I=";
+    hash = "sha256-1V95S0FWHzCxztgip+rbCjji4O71D+QdcSZ/hbABeKg=";
   };
 
   cargoLock = {
@@ -30,6 +30,13 @@ rustPlatform.buildRustPackage rec {
       "salsa-0.18.0" = "sha256-EjpCTOB6E7n5oNn1bvzNyznzs0uRJvAXrNsZggk4hgM=";
     };
   };
+
+  # Revert the change made in https://github.com/astral-sh/ruff/pull/13299
+  # It was causing linking issues: https://github.com/NixOS/nixpkgs/pull/341674#issuecomment-2351172084
+  postPatch = ''
+    substituteInPlace crates/ruff_benchmark/Cargo.toml \
+      --replace-fail '"unprefixed_malloc_on_supported_platforms"' ' '
+  '';
 
   nativeBuildInputs = [ installShellFiles ];
 

